pkg/utils/log.go

```diff
@@ -0,0 +1,27 @@
+package utils
+
+import (
+	"os"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// NewLogger returns a new zap.SugaredLogger
+func NewLogger() *zap.SugaredLogger {
+	var config zap.Config
+	debugMode, ok := os.LookupEnv("NUMAFLOW_DEBUG")
```

Comment:
was this environment variable something you imagined the user would specify in their container spec? If so, I wonder if we need a way for them to be able to accomplish the same thing if they use the user-friendly spec method.

Reply:
I think the env var "NUMAFLOW_DEBUG" is already supported on the numaflow side. https://numaflow.numaproj.io/development/debugging/

Reply:
So, [here](https://numaflow.numaproj.io/user-guide/reference/configuration/environment-variables/) it says that it applies to the "numa" container. That implies it's not being passed to the udf container - is it?

Reply:
Assuming it's not passed to the UDF container, then we need to figure out a way to pass it.

I am thinking of the following approach. How a UDSource determines whether to log at DEBUG mode is completely up to the UDSource implementer. Hence the following approach makes sense in a way that we can add to README.me to tell user, if you want to enable debug logging for nats-source-go, add this env var.

```
udsource:
    container:
          image: my-udsource:latest
          env:
            - name: NUMAFLOW_DEBUG
              value: "true"
```

For the user-friendly spec method, we need a way to pass it. I will dive deeper into how as I make changes to the controller.

Reply:
Okay, sounds good